src/util/buf.rs

```diff
@@ -0,0 +1,560 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use grpcio_sys::*;
+use std::ffi::c_void;
+use std::fmt::{self, Debug, Formatter};
+use std::io::{self, BufRead, Read};
+use std::mem::{self, ManuallyDrop, MaybeUninit};
+use std::ptr;
+use std::sync::atomic::{self, AtomicUsize, Ordering};
+
+/// A convenient rust wrapper for the type `grpc_slice`.
+#[repr(C)]
+pub struct GrpcSlice(grpc_slice);
```

Comment:
Can the user assume that the inner grpc_slice is always valid?

Reply:
Yes.

Reply:
Great! Could you add it to the docs please?